### PR TITLE
Add events for student notes

### PR DESCRIPTION
**Pull request by srpearce**
@lamagnifica Can you review?

### CODE REVIEW

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 1]
```diff
@@ -415,6 +415,8 @@ outside the Instructor Dashboard.
 
 * :ref:`pdf`
 
+* :ref:`student_notes`
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: I would tend to put this a little lower in the list, maybe after discussion forums events.

**Comment by lamagnifica** [comment]
on it. (for the future, it would be helpful to have links to the engineering PR or design page in the wiki here for reference)

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 2]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+**History**: This event was added in April 2015.
+
+``event`` **Member Fields**: 
+
+.. list-table::
+   :widths: 15 15 60
+   :header-rows: 1
+
+   * - Field
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: I would alphabetize these as there's no telling what order they'll be included in the event field (applies to all)

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 3]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+
+This section includes descriptions of the following events. 
+
+* ``edx.course.student_notes.added``
+* ``edx.course.student_notes.viewed``
+* ``edx.course.student_notes.edited``
+* ``edx.course.student_notes.deleted``
+* ``edx.course.student_notes.searched``
+* ``edx.course.student_notes.used_unit_link``
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: It might be useful to the readers to have a sentence or 2 about what student notes are, and a cross reference to where in the B&R guide they're described in full. (I started doing this for new categories of events about 6-8 months ago. See the library events for an example)

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 4]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
 
+.. _student_notes:
+
+=================================
+Student Notes Events   
+=================================
+
+This section includes descriptions of the following events. 
+
+* ``edx.course.student_notes.added``
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: I would put this list and the event descriptions in alphabetical order as this section is used for reference

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 5]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+.. list-table::
+   :widths: 15 15 60
+   :header-rows: 1
+
+   * - Field
+     - Type
+     - Details
+   * - ``note_id``
+     - string
+     - The ID of the note.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: I would state it if this ID is  unique. (throughout)

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 6]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+
+   * - Field
+     - Type
+     - Details
+   * - ``note_id``
+     - string
+     - The ID of the note. 
+   * - ``note_text``
+     - string
+     - The text of the note (truncated at 300 characters).
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: make the parenthetical statement a complete sentence: Notes are truncated to 300 characters. (throughout)

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 7]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+
+``edx.course.student_notes.added``
+***********************************
+
+The browser emits ``edx.course.student_notes.added`` events when a learner
+adds a note in courseware.
+
+**Event Source**: Browser
+
+**History**: This event was added in April 2015.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: provide the specific date if possible. For example, 
Added 5 May 2015; limited release of this feature began in April 2015.

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 8]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+     - Indicates whether the ``note_text`` field is truncated.
+   * - ``highlighted_content``
+     - string  
+     - The course text that the learner highlighted (truncated at 300 characters).
+   * - ``highlighted_content_truncated``
+     - boolean 
+     - Indicates whether the ``highlighted_content`` field is truncated.
+   * - ``component_usage_id``
+     - string  
+     - The ID of the component where the learner added the note.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: how is this used? That is, if I wanted to be able to tell what HTML component the learner was viewing when s/he made a note, I would look at the value in this field and compare it to what?

What I think you would do is take it and compare it to is the module identifiers for the HTML components in teh course_structure file... an example is shown in this section:
http://edx.readthedocs.org/projects/devdata/en/latest/internal_data_formats/course_structure.html#course-component-data but please confirm

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 9]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+
+``edx.course.student_notes.viewed``
+***********************************
+
+The browser emits ``edx.course.student_notes.viewed`` events when a learner
+views notes in courseware.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: does that mean, looks at an HTML component that happens to have a note? opens up a specific note while viewing an HTML component? or is this tracking when the student deliberately opens the Notes page?

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 10]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+   :widths: 15 15 60
+   :header-rows: 1
+
+   * - Field
+     - Type
+     - Details
+   * - ``notes``
+     - list
+     - A list of the currently visible notes. Learners can add multiple notes
+       to the same text.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: this might need to change based on what exacttly they're doing...

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 11]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+
+``event.notes`` **Member Fields**: 
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: no need to call this out in a separate table. Instead, after lines 1911-1912, add something like, "Contains the member field `note_id`, which contains the unique identifier for the note.

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 12]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+The browser emits ``edx.course.student_notes.edited`` events when a learner
+edits a note in courseware.
+
+**Event Source**: Browser
+
+**History**: This event was added in April 2015.
+
+``event`` **Member Fields**: 
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: I would make use of a cross reference back up to the `added` field here to minimize repetition. Take the 6 fields out of this table that are already described above and just describe the two "old" fields. Maybe something like the video events, such as http://edx.readthedocs.org/projects/devdata/en/latest/internal_data_formats/tracking_logs.html#load-video-edx-video-loaded

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 13]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+The browser emits ``edx.course.student_notes.deleted`` events when a learner
+deletes a note in courseware.
+
+**Event Source**: Browser
+
+**History**: This event was added in April 2015.
+
+``event`` **Member Fields**: 
+
+.. list-table::
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: replace the table with an xref up to the ``added` event

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 14]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+
+   * - Field
+     - Type
+     - Details
+   * - ``note_id``
+     - string
+     - The ID of the note.
+   * - ``note_text``
+     - string
+     - Text of the new note (truncated at 300 characters).
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: (not new, deleted)

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 15]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+     - Details
+   * - ``number_of_results``
+     - integer
+     - The number of search results.
+   * - ``search_string``
+     - string
+     - The text of the search query.
+
+``edx.course.student_notes.used_unit_link``
+*************************************
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: the heading markup has to be as long as the string above

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 16]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+
+The browser emits ``edx.course.student_notes.used_unit_link`` events when a
+learner clicks the link from a note on the **Student Notes** page to the
+location of the note in the courseware.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: "...clicks the link from a note on the **Student Notes** page to the location of the note in the courseware." --> something like, "...uses a note hyperlink on the **Student Notes** page to navigate to the HTML component with that note."

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 17]
```diff
@@ -1832,6 +1834,233 @@ user selects or clears the **Match Case** option for a search.
...
+
+   * - Field
+     - Type
+     - Details
+   * - ``note_id``
+     - string
+     - The ID of the note.
+   * - ``component_usage_id``
+     - string  
+     - The ID of the component where the learner added the note.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: this raises similar questions to line 1888...

**Comment by lamagnifica** [comment]
dev edit complete. great start, a few questions to investigate...

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 9 above)
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: @cahrens, can you answer this question?

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 18]
```diff
@@ -2815,6 +2816,194 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+.. list-table::
+   :widths: 15 15 60
+   :header-rows: 1
+
+   * - Field
+     - Type
+     - Details
+   * - ``component_usage_id``
+     - string  
+     - The ID of the component where the learner added the note.
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: @cahrens This is the unique ID, correct?

Also, how is this used? That is, if I wanted to be able to tell what HTML component the learner was viewing when s/he made a note, would I look at the value in this field and compare it to the module identifiers for the HTML components in the course_structure file?

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 19]
```diff
@@ -2815,6 +2816,194 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+
+``edx.course.student_notes.added``
+***********************************
+
+The browser emits ``edx.course.student_notes.added`` events when a learner
+adds a note in courseware.
+
+**Event Source**: Browser
+
+**History**: Added 12 May 2015.
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: @cahrens I'm assuming these will go out with next week's release; can you confirm?

**Comment by srpearce** [comment]
See also [Student Notes Events](https://openedx.atlassian.net/wiki/display/TNL/Student+Notes+Events)

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 20]
```diff
@@ -2815,6 +2816,245 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+     - The text of the note. Notes are truncated at 300 characters.
+   * - ``note_text_truncated``
+     - boolean
+     - Indicates whether the ``note_text`` field is truncated.
+   * - ``tags``
+     - array of strings
+     - An array of the tags that the learner has specified. The text is
+       truncated at 300 characters. If the text is truncated, only complete
+       tag names appear in the array. If the learner has not specified tags,
+       the array is empty.
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst:2885]: @lamagnifica Is this text clear? I was trying to think of a more elegant way to say "Array of tags (truncated after combined length of tag names > 300 characters). If no tags have been specified, will be an empty array. If truncated, truncation will not happen in the middle of tag (all tag names going over the limit will be left out of the array)."

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 21]
```diff
@@ -2815,6 +2816,245 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+     - Content of the array of tags before the learner edited it. Text is
+       truncated at 300 characters. If the text is truncated, only complete
+       tag names appear in the array. If the learner has not specified tags,
+       the array is empty.
+   * - ``old_tags_truncated``
+     - boolean
+     - Indicates whether the ``old_tags`` field is truncated.
+
+``edx.course.student_notes.notes_page_viewed``
+**********************************************
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst:2938]: @lamagnifica The new event - please review.

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 22]
```diff
@@ -2815,6 +2816,245 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+     - Indicates whether the ``old_note_text`` field is truncated.
+   * - ``old_tags``
+     - array of strings
+     - Content of the array of tags before the learner edited it. Text is
+       truncated at 300 characters. If the text is truncated, only complete
+       tag names appear in the array. If the learner has not specified tags,
+       the array is empty.
+   * - ``old_tags_truncated``
+     - boolean
+     - Indicates whether the ``old_tags`` field is truncated.
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: @lamagnifica `old_tags` and `old_tags_truncated` are the two new fields for this event - please review.

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 23]
```diff
@@ -2815,6 +2816,245 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+     - string
+     - The ID of the note. 
+   * - ``note_text``
+     - string
+     - The text of the note. Notes are truncated at 300 characters.
+   * - ``note_text_truncated``
+     - boolean
+     - Indicates whether the ``note_text`` field is truncated.
+   * - ``tags``
+     - array of strings
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst:2876]: @lamagnifica `tags` and `tags_truncated` are the two new member fields for this event - please review.

**Comment by srpearce** [comment]
@cahrens There are a few outstanding questions inline; can you please review them?

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 9 above)
Comment (cahrens) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: It is when they view the note on a unit page (actually bring up the annotator viewer).

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 18 above)
Comment (cahrens) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: It looks like this (in a course using split modulestore)--

'block-v1:course+christina+bar+type@html+block@cc128ccdb8604593a919cb2fb02c0115'

It is the unique identifier for the component within the course. I'm not sure what course_structure file you are referring to.

**Comment by cahrens** [comment]
Let me know if I missed anything, @srpearce.

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 18 above)
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: @cahrens, FYI the data packages include a course_structure file that provides just that -- IDs and relationships for all of the sections, subsections, units, and components. So the answer to @srpearce's question is yes!

Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: @lamagnifica So what should I use for text here? "The unique ID of the component where the learner added the note. This ID is the module identifier for the component in the course_structure file." - ?

Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: I'd keep it simple @srpearce, but give an xref, more like, 
The unique identifier for the HTML component where the learner added the note. For more information about the components that a course includes, see (:ref: to the course_structure.rst file)

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 24]
```diff
@@ -2815,6 +2816,237 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+     - string
+     - The ID of the note. 
+   * - ``note_text``
+     - string
+     - The text of the note. Notes are truncated at 8333 characters.
+   * - ``tags``
+     - array of strings
+     - An array of the tags that the learner has specified. The text is
+       truncated at 8333 characters. If the text is truncated, only complete
+       tag names appear in the array. If the learner has not specified tags,
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: has not specified --> did not specify

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 25]
```diff
@@ -2815,6 +2816,237 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+       the array is empty.
+   * - ``truncated``
+     - array of strings
+     - Lists the names of any truncated fields. Possible
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: So, the array can have one or more of the listed values? "possible" threw me here, making me infer that there might be others...

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 26]
```diff
@@ -2815,6 +2816,237 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+   :widths: 15 15 60
+   :header-rows: 1
+
+   * - Field
+     - Type
+     - Details
+   * - ``old_note_text``
+     - string
+     - The text of the note before the learner edited it. Notes are truncated
+       at 300 characters.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: is this field still limited to 300 characters?

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 27]
```diff
@@ -2815,6 +2816,237 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+   * - ``old_tags``
+     - array of strings
+     - Content of the array of tags before the learner edited it. Text is
+       truncated at 8333 characters. If the text is truncated, only complete
+       tag names appear in the array. If the learner has not specified tags,
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: same as above

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 28]
```diff
@@ -2815,6 +2816,237 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+       the array is empty.
+
+``edx.course.student_notes.notes_page_viewed``
+**********************************************
+
+The browser emits ``edx.course.student_notes.notes_page_viewed`` events when a
+student accesses or selects a different view on the **Notes** page. Note that
+because "Recent Activity" is the default view, the browser always emits a
+``edx.course.student_notes.notes_page_viewed`` event that has a ``view`` field
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst:2943]: a --> an

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 29]
```diff
@@ -2815,6 +2816,237 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+     - Details
+   * - ``component_usage_id``
+     - string  
+     - The ID of the component where the learner added the note.
+   * - ``note_id``
+     - string
+     - The ID of the note.
+   * - view
+     - string
+     - When a learner selects a different view on the Notes page, the view
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: bold Notes

Actually, is this capturing which of the Notes pages the learner was on when they clicked a link to navigate to the part of the course where they made the note? If so, I'd reword a bit to make that clearer. This sounds like they're just investigating around in the Notes tab.

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 30]
```diff
@@ -2815,6 +2816,237 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+     - The number of search results.
+   * - ``search_string``
+     - string
+     - The text of the search query.
+
+``edx.course.student_notes.used_unit_link``
+*********************************************
+
+The browser emits ``edx.course.student_notes.used_unit_link`` events when a
+learner uses a note link on the **Student Notes** page to go to the HTML
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: is it Student Notes or just Notes?

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 31]
```diff
@@ -2815,6 +2816,237 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+
+   * - Field
+     - Type
+     - Details
+   * - ``notes``
+     - list
+     - A list of the ``note_id`` values for the currently visible notes.
+       Learners can add multiple notes to the same text.
+
+Contains the member field ``note_id``, which contains the unique identifier for the note.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: should this be indented? not clear to me what field contains this note_id member field, and what relationship it has to the notes list described on 3045-3047

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 32]
```diff
@@ -2815,6 +2816,237 @@ complete, the server emits an ``edx.forum.thread.created`` event.
...
+     - string
+     - When a learner selects a different view on the Notes page, the view
+       that the learner last viewed. Possible values are "Recent Activity",
+       "Location in Course", "Tags", and "Search Results".
+
+``edx.course.student_notes.viewed``
+***********************************
+
+The browser emits ``edx.course.student_notes.viewed`` events when a learner
+selects a note to view the note in the courseware.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: maybe, "...selects highlighted text in the courseware to view the associated note"?

File: en_us/data/source/front_matter/change_log.rst [hunk 33]
```diff
@@ -14,6 +14,19 @@ January-March 2015
 
    * - Date
      - Change
```
Comment (lamagnifica) [inline en_us/data/source/front_matter/change_log.rst:16]: this should be in an April-June table! (see line 8)

**Comment by lamagnifica** [comment]
this is looking good! a took another look after seeing changes to the truncation limit go by

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 25 above)
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: Would it be more clear to say "Values can be..."?

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 29 above)
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: The user *is* just clicking different views on the **Notes** page. :-)

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 32 above)
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: Not quite - this happens on the **Notes** page, which lists the notes. A learner clicks a note in the list and is taken to the note inside the courseware. I'll make that clearer.

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 31 above)
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: I'm confused - this was actually a change you told me to make in the first round of edits. Originally I had two member fields and their associated tables here, the way it appears in the spec (https://openedx.atlassian.net/wiki/display/TNL/Student+Notes+Events) - ?

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 32 above)
Comment (cahrens) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: No, this event is emitted when a user views the note on the courseware page. One way to get to a note (and view it) is to click on the unit link on the Notes page, but another way is to just be on the unit in the courseware and select/click on the highlighted text.

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 29 above)
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: OK, I re-read the spec, and honestly I'm not sure what's happening here. Will investigate.

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 32 above)
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: Oh. yikes - will fix. Sigh.

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 29 above)
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: OK, clarified.

File: en_us/data/source/internal_data_formats/tracking_logs.rst (hunk 31 above)
Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: Wait, I think I've got it:

`event` **Member Fields**: 

.. list-table::
   :widths: 15 15 60
   :header-rows: 1
- \- Field
  - Type
  - Details
- \- `notes`
  - list
  - A list of the `note_id` values for the currently visible notes.
    Learners can add multiple notes to the same text. Contains the member
    field `note_id`, which contains the unique identifier for the note.

Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: ok, I found my original edit. "no need to call this out in a separate table. Instead, after lines 1911-1912, add something like, "Contains the member field note_id, which contains the unique identifier for the note."

my intention was that you might replace the description in the table cell for the `notes` field -- the sentence on line 3046. 

Does that make more sense?

Comment (srpearce) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: I _think_ that's the change I just made a second ago (see comment directly above) - is that description now redundant? My brain is seriously fried.

**Comment by srpearce** [comment]
Superseded by [PR 873](https://github.com/edx/edx-documentation/pull/873).
